packages/service-provider-core/src/admin.ts

```diff
@@ -116,4 +129,14 @@ export default interface Admin {
    * The FLE options passed to the client, if any.
    */
   getFleOptions?: () => AutoEncryptionOptions | undefined;
+
+  /**
+   * The helper method to correctly access FLE implementation's createEncryptedCollection
+   */
+  createEncryptedCollection?(
+    dbName: string,
+    collName: string,
+    options: CreateEncryptedCollectionOptions,
+    libmongocrypt: MongoCryptClientEncryption
```

Comment:
`libmongocrypt` is the same as `this.fle` in the service provider, so I would probably not pass it back to a method on the server provider? If they’re different, that would be a sign that something’s very broken

Reply:
Oh but this is an instance of libmongocrypt's ClientEncryption and what ServiceProviders have is just the imported class itself. From what I understood, the instance is created only by `Mongo.getClientEncryption` and also cached there on the Mongo instance. Since an instance was already in scope I decided to use it otherwise I would be doing something similar to `Mongo.getClientEncryption` in the service providers as well.